apps/election-manager/src/lib/votecounting.test.ts

```diff
@@ -500,6 +500,64 @@ describe('filterTalliesByParams in a primary election', () => {
   })
 })
 
+test('undervotes counted in n of m contest properly', () => {
+  // Create mock CVR data
+  const mockCVR: CastVoteRecord = {
+    _ballotId: 'abc',
+    _ballotStyleId: '12D',
+    _ballotType: 'standard',
+    _precinctId: '21',
+    _testBallot: false,
+    _scannerId: '1',
+    'county-commissioners': [],
+  }
+
+  // tabulate it
+  let electionTally = computeFullElectionTally(primaryElectionSample, [
+    [mockCVR],
+  ])!
+
+  // The county commissioners race has 4 seats. Each vote less then 4 should be counted
+  // as an additional undervote.
+  expect(
+    electionTally.overallTally.contestTallyMetadata['county-commissioners']!
+      .undervotes
+  ).toBe(4)
+
+  mockCVR['county-commissioners'] = ['argent']
+  electionTally = computeFullElectionTally(primaryElectionSample, [[mockCVR]])!
+  expect(
+    electionTally.overallTally.contestTallyMetadata['county-commissioners']!
+      .undervotes
+  ).toBe(3)
+
+  mockCVR['county-commissioners'] = ['argent', 'bainbridge']
+  electionTally = computeFullElectionTally(primaryElectionSample, [[mockCVR]])!
+  expect(
+    electionTally.overallTally.contestTallyMetadata['county-commissioners']!
+      .undervotes
+  ).toBe(2)
+
+  mockCVR['county-commissioners'] = ['argent', 'bainbridge', 'hennessey']
+  electionTally = computeFullElectionTally(primaryElectionSample, [[mockCVR]])!
+  expect(
+    electionTally.overallTally.contestTallyMetadata['county-commissioners']!
+      .undervotes
+  ).toBe(1)
+
+  mockCVR['county-commissioners'] = [
+    'argent',
+    'bainbridge',
+    'hennessey',
+    'savoy',
+  ]
+  electionTally = computeFullElectionTally(primaryElectionSample, [[mockCVR]])!
+  expect(
+    electionTally.overallTally.contestTallyMetadata['county-commissioners']!
+      .undervotes
+  ).toBe(0)
+})
```

Comment:
This style of test wouldn't work well with my changes to make things more read-only in #211. Perhaps try this instead?
```suggestion
test('undervotes counted in n of m contest properly', () => {
  // Create mock CVR data
  const mockCVR: CastVoteRecord = {
    _ballotId: 'abc',
    _ballotStyleId: '12D',
    _ballotType: 'standard',
    _precinctId: '21',
    _testBallot: false,
    _scannerId: '1',
    'county-commissioners': [],
  }

  // The county commissioners race has 4 seats. Each vote less then 4 should be counted
  // as an additional undervote.
  expect(
    computeFullElectionTally(primaryElectionSample, [[mockCVR]])?.overallTally
      .contestTallyMetadata['county-commissioners']?.undervotes
  ).toBe(4)

  expect(
    computeFullElectionTally(primaryElectionSample, [
      [{ ...mockCVR, 'county-commissioners': ['argent'] }],
    ])?.overallTally.contestTallyMetadata['county-commissioners']?.undervotes
  ).toBe(3)

  expect(
    computeFullElectionTally(primaryElectionSample, [
      [{ ...mockCVR, 'county-commissioners': ['argent', 'bainbridge'] }],
    ])?.overallTally.contestTallyMetadata['county-commissioners']?.undervotes
  ).toBe(2)

  expect(
    computeFullElectionTally(primaryElectionSample, [
      [
        {
          ...mockCVR,
          'county-commissioners': ['argent', 'bainbridge', 'hennessey'],
        },
      ],
    ])?.overallTally.contestTallyMetadata['county-commissioners']?.undervotes
  ).toBe(1)

  expect(
    computeFullElectionTally(primaryElectionSample, [
      [
        {
          ...mockCVR,
          'county-commissioners': [
            'argent',
            'bainbridge',
            'hennessey',
            'savoy',
          ],
        },
      ],
    ])?.overallTally.contestTallyMetadata['county-commissioners']?.undervotes
  ).toBe(0)
})
```